### ISSUE TITLE
Deprecate limiting, replace with Opaque Federation

### ISSUE THREAD

**Issue by SoniEx2** [mastodon/mastodon#22439]
### Pitch

Opaque Federation would make limiting unnecessary. Just ban them instead.

### Motivation

Limiting an user or an instance is usually done so the user/instance can keep their friends and stuff. It's basically recognizing a threat but not acting on it. Like sure, maybe the threat isn't an active threat anymore, or it's low-risk for some other reason. However, it provides a terrible experience for local users, especially because new follows still go through.

On the other hand there's Opaque Federation. Opaque Federation sidesteps the instance entirely, and allows users to build up connections to users their instance is not federated with. Compared to other social media, it's basically the equivalent of moving to Telegram after getting banned everywhere else. Except integrated into your local social media.

Among other benefits, it makes banning users easier, as admins no longer have to answer questions like "how much of a risk is this user to our other users for us to cut them off of their friends", which improves moderation. Because the answer is: you can't cut them off. Just ban them. No need to worry about it.

In other words, it makes it much easier to cut folks off your instance, without cutting them off their friends on your instance. This one aspect is a huge improvement over limiting, and is why we see it as a full replacement to limiting.

(Could that be done without Opaque Federation? Maybe! Mastodon really doesn't have any equivalent moderation tools tho.)

**Comment by ClearlyClaire** [comment]
Mastodon has no such thing as “opaque federation” nor any plan for it, it's a completely different architecture and security model as what Mastodon currently provides.
